pkg/storegateway/bucket.go

```diff
@@ -532,6 +534,9 @@ type seriesChunks struct {
 
 // Series implements the storepb.StoreServer interface.
 func (s *BucketStore) Series(req *storepb.SeriesRequest, srv storepb.Store_SeriesServer) (err error) {
+	if req.SkipChunks {
+		req.StreamingChunksBatchSize = 0
+	}
```

Comment:
Even if we take care of it on querier side, I would still prefer having this safety net here since there are different components that might talk to storegateway. I will change the logic at other places to just check `StreamingChunksBatchSize`.